### ISSUE TITLE
SVT:Hybrid:Appsody node red sample fails to build and start.

### ISSUE THREAD

**Issue by jagraj** [eclipse/codewind#2872]
**Codewind version:** 0.12.0
**OS:** Windows/OCP 4.3.18

**Che version:**
**IDE extension version:** 0.12.0
**IDE version:** Eclipse 2019-09 
**Kubernetes cluster:** OCP 4.3.18

**Description:**
Failed to build Appsody node red sample application on remote Codewind. This is the failure that I am getting.

When I go and look into app deployment I see this for pod.

**Pogs and their logs**

```
cw-appsodynodejslbcw012svt-775a4ea0-9072-11ea-989e-65bcd88jkwxj   0/1     CrashLoopBackOff   6          8m35s
cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e-5f84b5b2c7v9   0/1     CrashLoopBackOff   6          8m36s

```

```
oc logs cw-appsodynodejslbcw012svt-775a4ea0-9072-11ea-989e-65bcd88jkwxj
Running APPSODY_PREP command: npm install --prefix user-app && npm run build --prefix user-app
npm ERR! correctMkdir failed to make directory /.npm/_locks
npm WARN nodejs-loopback-scaffold@1.0.0 No license field.

npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /.npm
npm ERR! errno -13
npm ERR!
npm ERR! Your cache folder contains root-owned files, due to a bug in
npm ERR! previous versions of npm which has since been addressed.
npm ERR!
npm ERR! To permanently fix this problem, please run:
npm ERR!   sudo chown -R 1000580000:0 "/.npm"
[Error] FATAL error APPSODY_PREP command received an error.  The controller is exiting: exit status 243

```

**Logs from IDE**
```
Running development environment...
Pulling docker image docker.io/appsody/node-red:0.1
Running command: buildah pull docker.io/appsody/node-red:0.1
b85159adbdc7106cf26bc84498a65fc29283722896cbe7524ffbd782d0de723f
Successfully added your project to /codewind-workspace/.appsody/project.yaml
Attempting to apply resource in Kubernetes ...
Running command: kubectl apply -f /codewind-workspace/appsodynodejsredcw012svtocp43/app-deploy.yaml
Attempting to apply resource in Kubernetes ...
Running command: kubectl apply -f /codewind-workspace/appsodynodejsredcw012svtocp43/app-service.yaml
Getting the logs ...
Running command: kubectl logs deployment/cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e -f "--pod-running-timeout=2m"
[Container] Error from server (BadRequest): container "cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e" in pod "cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e-5f84b5b2c7v9" is waiting to start: PodInitializing
Getting the logs ...
Running command: kubectl logs deployment/cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e -f "--pod-running-timeout=2m"
[Container] Error from server (BadRequest): container "cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e" in pod "cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e-5f84b5b2c7v9" is waiting to start: PodInitializing
Getting the logs ...
Running command: kubectl logs deployment/cw-appsodynodejsredcw012sv-8ff374f0-9072-11ea-989e -f "--pod-running-timeout=2m"
[Container] Running APPSODY_PREP command: npm install --prefix user-app
[Container] npm ERR! correctMkdir failed to make directory /.npm/_locks
[Container] npm WARN SampleAppsody-Node-RED-Project@0.1.0 No repository field.
[Container] npm WARN SampleAppsody-Node-RED-Project@0.1.0 No license field.
[Container] 
[Container] npm ERR! code EACCES
[Container] npm ERR! syscall mkdir
[Container] npm ERR! path /.npm
[Container] npm ERR! errno -13
[Container] npm ERR! 
[Container] npm ERR! Your cache folder contains root-owned files, due to a bug in
[Container] npm ERR! previous versions of npm which has since been addressed.
[Container] npm ERR! 
[Container] npm ERR! To permanently fix this problem, please run:
[Container] npm ERR!   sudo chown -R 1000580000:0 "/.npm"
[Container] [Error] FATAL error APPSODY_PREP command received an error.  The controller is exiting: exit status 243
```

**Steps to reproduce:**
1. Install Codewind 0.12.0 on remote OCP 4.3 cluster.
2. Create Appsody Node Red application from the template.
3. You will get into above failure.

**Workaround:**

**Comment by jagraj** [comment]
I see same problem with Appsody Nodejs loop back application as well.

**Comment by makandre** [comment]
I wasn't able to reproduce this.  Both project types started fine for me.

**Comment by jagraj** [comment]
@tobespc @markcor11 @makandre 
Ok, After lot of debugging on various clusters with operator install and legacy way installing remote codewind makes big difference. The operator install not enabling OpenShift clusters for root access and privileged. Due to this all the templates need root access and they all failing. The operator install for OpenShift clusters must enable these configs for the apps to work.

`oc adm policy add-scc-to-group anyuid system:serviceaccounts:codewind`
`oc adm policy add-scc-to-group privileged system:serviceaccounts:codewind`

**Before enabling above configs you can see the failures under "ocp43-caliver" connection.**

![Screen Shot 2020-05-07 at 2 35 08 PM](https://user-images.githubusercontent.com/713849/81333083-0a32e300-9072-11ea-9b9c-9cce953f33e5.png)

**After enabling above configs you can see the failures under "ocp43-caliver" connection.**

![image](https://user-images.githubusercontent.com/713849/81333259-4fefab80-9072-11ea-8238-76bf7381310b.png)

With this info, I am assigning this defect to portal team.

**Comment by johnmcollier** [comment]
FYI, it seems the operator only enables the `privileged` policy, but not `anyuid`: https://github.com/eclipse/codewind-operator/blob/d10e28ab9fe90f15452bc7a0b9d32c3af3b84ba8/pkg/controller/codewind/rbac.go#L29

**Comment by markcor11** [comment]
What is the reason behind appsody projects needing the `anyuid` permission ?

**Comment by makandre** [comment]
@markcor11 Might be better answered by @neeraj-laad as to why Appsody needs `anyuid`

I know they have to spin up pods and services and they need to mount the codewind workspace pvc as well.

**Comment by jagraj** [comment]
Based on my understanding, some of our default basic templates also need ```anyuid``` policy to be  configured.

**Comment by johnmcollier** [comment]
Doesn't Codewind still run as root? Doesn't it need the `anyuid` permission too?

**Comment by markcor11** [comment]
@johnmcollier that change should be sufficient since there is an existing role binding between the service account (one per Codewind PFE) and the cluster roles containing the fix.

<img width="536" alt="Screenshot 2020-05-11 at 13 12 44" src="https://user-images.githubusercontent.com/28102564/81613920-2cf92a80-93d7-11ea-9d3d-dd1cfc660393.png">

**Comment by jagraj** [comment]
This issue has been fixed with the latest 0.12.0 operator install. Thanks for the fix.
